newIDE/app/src/EventsSheet/EventsTree/Renderers/CommentEvent.js

```diff
@@ -1,83 +1,78 @@
 // @flow
+import { t } from '@lingui/macro';
+
 import * as React from 'react';
-import ReactDOM from 'react-dom';
 import classNames from 'classnames';
-import { type EventRendererProps } from './EventRenderer';
-import { rgbToHex } from '../../../Utils/ColorTransformer';
+import TextField from '../../../UI/TextField';
 import {
   largeSelectedArea,
   largeSelectableArea,
   selectableArea,
+  disabledText,
 } from '../ClassNames';
+import { type EventRendererProps } from './EventRenderer';
 const gd = global.gd;
 
 const styles = {
   container: {
-    minHeight: 30,
     display: 'flex',
-    backgroundColor: '#fbf3d9',
-  },
-  text: {
-    flex: 1,
-    whiteSpace: 'pre-line',
-    margin: 0,
+    flexWrap: 'wrap',
     padding: 5,
+    overflow: 'hidden',
   },
-  textArea: {
-    padding: 5,
-    flex: 1,
-    boxSizing: 'border-box',
+  textField: {
     width: '100%',
     fontSize: 14,
   },
 };
 
-type State = {|
-  editing: boolean,
-  height: number,
-|};
-
 export default class CommentEvent extends React.Component<
   EventRendererProps,
-  State
+  *
 > {
   state = {
     editing: false,
-    height: 0,
+    height: '100%',
   };
 
-  _container: ?any;
-  _input: ?any;
+  _selectable: ?any;
+  _textField: ?any;
 
   edit = () => {
-    if (!this._container) return;
-
     this.setState(
       {
         editing: true,
-        height: this._container.offsetHeight,
+        height: '100%',
       },
       () => {
-        // $FlowFixMe
-        const input: ?HTMLInputElement = ReactDOM.findDOMNode(this._input);
-        if (input) {
-          input.focus();
-          input.value = gd.asCommentEvent(this.props.event).getComment();
-        }
+        if (this._textField) this._textField.focus();
+        this.forceUpdate();
       }
     );
   };
 
-  endEditing = () => {
+  onEvent = (e: any, text: string) => {
     const commentEvent = gd.asCommentEvent(this.props.event);
+    commentEvent.setComment(text);
 
-    // $FlowFixMe
-    const input: ?HTMLInputElement = ReactDOM.findDOMNode(this._input);
-    if (input) commentEvent.setComment(input.value);
+    this.setState(
```

Comment:
You should be able to remove this setState and just do:

```
        this.props.onUpdate();
        this.forceUpdate();
```

onUpdate to let the parent know that the event was modified (it's a requirement)
forceUpdate because we modified something in libGD.js (`setComment`) that should be reflected in the interface of this component (React would not re-render the component otherwise).